packages/server/graphql/private/mutations/processRecurrence.ts

```diff
@@ -0,0 +1,124 @@
+import MeetingTeamPrompt from '../../../database/types/MeetingTeamPrompt'
+import {getActiveMeetingSeries} from '../../../postgres/queries/getActiveMeetingSeries'
+import {MeetingSeries} from '../../../postgres/queries/getMeetingSeriesByIds'
+import {analytics} from '../../../utils/analytics/analytics'
+import publish from '../../../utils/publish'
+import standardError from '../../../utils/standardError'
+import isStartMeetingLocked from '../../mutations/helpers/isStartMeetingLocked'
+import {IntegrationNotifier} from '../../mutations/helpers/notifications/IntegrationNotifier'
+import safeCreateTeamPrompt from '../../mutations/helpers/safeCreateTeamPrompt'
+import safeEndTeamPrompt from '../../mutations/helpers/safeEndTeamPrompt'
+import {MutationResolvers} from '../resolverTypes'
+
+const startRecurringTeamPrompt = async (
+  meetingSeries: MeetingSeries,
+  startTime: Date,
+  dataLoader,
+  r,
+  subOptions
+) => {
+  const {teamId, facilitatorId} = meetingSeries
+
+  // AUTH
+  const unpaidError = await isStartMeetingLocked(teamId, dataLoader)
+  if (unpaidError) return standardError(new Error(unpaidError), {userId: facilitatorId})
+
+  const formattedDate = new Date().toLocaleDateString('en-US', {
+    month: 'short',
+    day: 'numeric'
+  })
+
+  const meeting = await safeCreateTeamPrompt(teamId, facilitatorId, r, dataLoader, {
+    name: `${meetingSeries.title} - ${formattedDate}`,
+    scheduledEndTime: new Date(startTime.getTime() + ms(`${meetingSeries.duration}m`)),
+    meetingSeriesId: meetingSeries.id
+  })
+
+  meeting.name = `${meetingSeries.title} - ${formattedDate}`
+  meeting.scheduledEndTime = new Date(startTime.getTime() + ms(`${meetingSeries.duration}m`))
+  meeting.meetingSeriesId = meetingSeries.id
+
+  await r.table('NewMeeting').insert(meeting).run()
+
+  IntegrationNotifier.startMeeting(dataLoader, meeting.id, teamId)
+  analytics.meetingStarted(facilitatorId, meeting)
+  const data = {teamId, meetingId: meeting.id}
+  publish(SubscriptionChannel.TEAM, teamId, 'StartTeamPromptSuccess', data, subOptions)
+}
+
+const processRecurrence: MutationResolvers['processRecurrence'] = async (
+  _source,
+  {},
+  {authToken, dataLoader, socketId: mutatorId}
+) => {
+  const r = await getRethink()
+  const now = new Date()
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+
+  // VALIDATION
+
+  // RESOLUTION
+  // Find any meetings with a scheduledEndTime before now, and close them
+  const teamPromptMeetingsToEnd = (await r
+    .table('NewMeeting')
+    .between([false, r.minval], [false, now], {index: 'hasEndedScheduledEndTime'})
+    .filter({meetingType: 'teamPrompt'})
+    .run()) as MeetingTeamPrompt[]
+
+  const res = await Promise.all(
+    teamPromptMeetingsToEnd.map(async (meeting) => {
+      return await safeEndTeamPrompt({meeting, now, dataLoader, r, subOptions})
+    })
+  )
+
+  const meetingsEnded = res.filter((res) => !('error' in res)).length
+
+  let meetingsStarted = 0
```

Comment:
+1 would like to see this as `const` like up above

Reply:
Changing this is less trivial compared to `meetingsEnded`, since `meetingsStarted` is updated in a nested loop, so getting this as a const would probably require a map-reduce-y thing, which I think would be less readable compared to incrementing a `let` var. WDYT?

Reply:
sure, makes sense